errdefs/notfound.go

```diff
@@ -0,0 +1,69 @@
+package errdefs
+
+import (
+	"errors"
+	"fmt"
+)
+
+// NotFound is an error interface which denotes whether the opration failed due
+// to a the resource not being found.
+type ErrNotFound interface {
+	NotFound() bool
+	error
+}
+
+type notFoundError struct {
+	cause error
+}
+
+func (e *notFoundError) Error() string {
+	return e.cause.Error()
+}
+
+func (e *notFoundError) NotFound() bool {
+	return true
+}
+
+func (e *notFoundError) Cause() error {
+	return e.Cause()
+}
+
+// AsNotFound wraps the passed in error to makeit of type ErrNotFound
+//
+// Callers should make sure the passed in error has exactly they error message
+// it wants as this function does not decorate the message.
+func AsNotFound(err error) error {
+	if err == nil {
+		return nil
+	}
+	return &notFoundError{err}
+}
+
+// NotFound makes an ErrNotFound from the provided error message
+func NotFound(msg string) error {
+	return &notFoundError{errors.New(msg)}
+}
+
+// NotFoundf makes an ErrNotFound from the provided error format and args
+func NotFoundf(format string, args ...interface{}) error {
+	return &notFoundError{fmt.Errorf(format, args...)}
+}
+
+// IsNotFound determines if the passed in error is of type ErrNotFound
+//
+// This will traverse the causal chain (`Cause() error`), until it finds an error
+// which implements the `NotFound` interface.
+func IsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	if e, ok := err.(ErrNotFound); ok {
+		return e.NotFound()
+	}
+
+	if e, ok := err.(Causal); ok {
```

Comment:
Refer to other comment about pkg/errors.

Reply:
Also, why should Casual be a public interface?

Reply:
I think I made it public for documentation purposes.

Reply:
But it may not be necessary.

Reply:
It's awkward right -- if someone wants to write their own wrap methods which may eventually drill-down to a errdefs error, they want to know how to implement this interface. Do we want people doing that?  

The route that pkgerrors goes is that it defines the interface in the docstring, which I think is good enough.